src/desc.rs

```diff
@@ -77,15 +78,24 @@ impl Desc {
             return Err(Error::Msg("empty help string".into()));
         }
 
-        // TODO: check valid metric fq_name.
+        lazy_static! {
+            static ref VALID_METRIC_NAME: Regex = Regex::new(r"^[a-zA-Z_][a-zA-Z0-9_:]*$").unwrap();
+            static ref VALID_LABEL_NAME: Regex = Regex::new(r"^[a-zA-Z_][a-zA-Z0-9_]*$").unwrap();
+        }
+
+        if !VALID_METRIC_NAME.is_match(&desc.fq_name) {
+            return Err(Error::Msg(format!("'{}' is not a valid metric name", desc.fq_name)))
+        }
 
         let mut label_values = Vec::with_capacity(const_labels.len() + 1);
         label_values.push(fq_name);
 
         let mut label_names = BTreeSet::new();
 
         for label_name in const_labels.keys() {
-            // TODO: check invalid label name
+            if !VALID_LABEL_NAME.is_match(&label_name) {
+                return Err(Error::Msg(format!("'{}' is not a valid label name", &label_name)))
```

Comment:
the Makefile doesn't have a `format` target. I've added semicolons and newlines manually.